docs/design-documents/platform/crash-reporting/crash_reporting.md

```diff
@@ -0,0 +1,286 @@
+# Crash Reporting
+
+# Table of contents
+
+- [Crash Reporting](#crash-reporting)
+- [Table of contents](#table-of-contents)
+    - [Revision history](#revision-history)
+- [Introduction](#introduction)
+    - [Overview and background](#overview-and-background)
+    - [Requirements and assumptions](#requirements-and-assumptions)
+- [System architecture and high-level design](#system-architecture-and-high-level-design)
+    - [System architecture and component interaction](#system-architecture-and-component-interaction)
+- [Detailed design](#detailed-design)
+- [Usage scenarios and examples](#usage-scenarios-and-examples)
+- [Tools and configuration changes](#tools-and-configuration-changes)
+- [Other information](#other-information)
+    - [Reusability](#reusability)
+    - [Deprecations](#deprecations)
+    - [References](#references)
+
+### Revision history
+
+1.0 - Initial version - Senthil Ramakrishnan - 10/22/2018
+
+# Introduction
+
+### Overview and background
+
+MbedOS currently implements error/exception handlers which gets invoked when the system encounters a fatal error/exception. The error handler capture information such as register context/thread info etc and these are valuable information required to debug the problem later. This information is currently printed over the serial port, but in many cases the serial port is not accessible and the serial terminal log is not captured, particularly in the case of field deployed devices. We cannot save this information by sending it over network or writing to a file, as the state of the system might be unstable after the fatal error. And thus a different mechanism is needed to record and report this data. The idea here is to auto-reboot the system after a fatal error has occurred to bring the system back in stable state, without losing the RAM contents where we have the error information collected, and we can then save this information reliably to be logged externally(E.g:- ARM Pelion cloud) or can be written to file system. 
+
+### Requirements and assumptions
+
+This feature requires 256 bytes of dedicated RAM allocated for storing the error and fault context information.
```

Comment:
Allocated where? When? How?
A single line for requirements and assumptions feels really light to me.